src/main/diagnostics/diagnostics.ts

```diff
@@ -54,9 +55,40 @@ export class Diagnostics {
   /**
    * @param categoryName {string}
    * @param id {string}
-   * Returns {unknown} if the categoryName isn't known or id not in that category, the check object otherwise.
+   * Returns an array of all matching checkObjects, depending on which of categoryName and id are specified.
    */
-  getCheckByID(categoryName: string, id: string): DiagnosticsCheck {
-    return this.checksByCategory[categoryName]?.filter(check => check.id === id)[0];
+  getChecks(categoryName: string|null, id: string|null): DiagnosticsCheck[] {
```

Comment:
This would be cleaner if we just had a list of all checks, and simply did:
```ts
return this.checks
  .filter(check => categoryName ? check.category === categoryName : true)
  .filter(check => id ? check.id === id : true);
```

Reply:
done